### PR TITLE
enhancement: enable array API support in DBSCAN

### PR DESCRIPTION
**Pull request by icfaust**
## Description

This enables array API dispatching for the DBSCAN algorithm. This required the following changes:
 
- removed array checking and raw inputs interaction in onedal.cluster.dbscan.py
- removed related imports
- moved sklearn conformance of ```components_``` to sklearnex
- Simplified spmd interface to minimize maintenance
- generalize working namespace to ```get_namespace```
- move dtype checking and finite checking to use ```validate_data```
- Add aspects of #2566 for enabling the array API tags

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

File: onedal/cluster/dbscan.py [hunk 1]
```diff
@@ -64,37 +60,21 @@ def _get_onedal_params(self, dtype=np.float32):
...
         else:
-            # construct keyword arguments for different namespaces (dptcl takes sycl_queue)
-            kwargs = {"dtype": xp.int32}  # always the same
-            if xp is not np:
-                kwargs["sycl_queue"] = queue
-            self.core_sample_indices_ = xp.empty((0,), **kwargs)
-        self.components_ = xp.take(X, self.core_sample_indices_, axis=0)
-        self.n_features_in_ = X.shape[1]
+            self.core_sample_indices_ = None
         return self
```
[the remark this answers is not in the document]
Comment (icfaust) [reply onedal/cluster/dbscan.py:80]: Good catch, but these attributes are sklearn conformance, and do not belong in this estimator.

File: sklearnex/utils/_array_api.py [hunk 2]
```diff
@@ -80,3 +85,38 @@ def get_namespace(*arrays):
         return sklearn_get_namespace(*arrays)
     else:
         return np, False
+
+
+def enable_array_api(original_class: type[oneDALEstimator]) -> type[oneDALEstimator]:
```
[the remark this answers is not in the document]
Comment (icfaust) [reply sklearnex/utils/_array_api.py]: Good thing we are 3.9+

File: sklearnex/cluster/dbscan.py [hunk 3]
```diff
@@ -86,9 +74,18 @@ def __init__(
         self.p = p
         self.n_jobs = n_jobs
 
+    _onedal_dbscan = staticmethod(onedal_DBSCAN)
+
     def _onedal_fit(self, X, y, sample_weight=None, queue=None):
-        if get_config()["use_raw_input"] is False:
-            X = validate_data(self, X, ensure_all_finite=False)
+        xp, _ = get_namespace(X, y, sample_weight)
```
[the remark this answers is not in the document]
Comment (icfaust) [reply sklearnex/cluster/dbscan.py:80]: This may conflict with #2589, I'd say leave it and then when the that PR is applied we can address this.

File: onedal/cluster/dbscan.py [hunk 4]
```diff
@@ -64,37 +60,21 @@ def _get_onedal_params(self, dtype=np.float32):
...
+            )[:, 0]
         else:
-            # construct keyword arguments for different namespaces (dptcl takes sycl_queue)
-            kwargs = {"dtype": xp.int32}  # always the same
-            if xp is not np:
-                kwargs["sycl_queue"] = queue
-            self.core_sample_indices_ = xp.empty((0,), **kwargs)
-        self.components_ = xp.take(X, self.core_sample_indices_, axis=0)
-        self.n_features_in_ = X.shape[1]
+            self.core_sample_indices_ = None
```
[the remark this answers is not in the document]
Comment (icfaust) [reply onedal/cluster/dbscan.py:79]: This will make the return types mixed anyway, setting to None is more general. No thanks copilot.

**Review by Vika-F — APPROVED** [review]
All the questions that I wanted to ask (about the move of attributes from onedal to sklearnex; and the core_sample_indices_ = None) were already asked by Copilot and the answers are provided.

LGTM
